PxWeb/Code/Api2/DataSelection/SelectionHandler.cs

```diff
@@ -28,50 +33,80 @@ public class SelectionHandler : ISelectionHandler
+        /// <returns>If everything was ok, an array of selection objects, else null</returns>
+        public Selection[]? GetSelection(IPXModelBuilder builder, PXModel model, VariablesSelection? variablesSelection, out Problem? problem)
         {
+            if (!VerifyAndApplyCodelists(builder, model, variablesSelection, out problem))
+            {
+                return null;
+            }
+
+            Selection[]? selections;
+
             if  (variablesSelection is not null && HasSelection(variablesSelection))
             {
                 //Add variables that the user did not post
                 variablesSelection = AddVariables(variablesSelection, model);
 
                 //Map VariablesSelection to PCaxis.Paxiom.Selection[] 
-                return MapCustomizedSelection(model, variablesSelection).ToArray();
+                selections = MapCustomizedSelection(builder, model, variablesSelection).ToArray();
             }
             else
             {
-                return GetDefaultSelection(model);
+                selections = GetDefaultSelection(model);
             }
 
+            if (!CheckNumberOfCells(selections))
+            {
+                selections = null;
+                problem = TooManyCellsSelected();
+            }
+
+            return selections;
+
         }
 
         /// <summary>
-        /// Verify that VariablesSelection object has valid variables and values
+        /// Verify that VariablesSelection object has valid variables and values. Also applies codelists.
         /// </summary>
+        /// <param name="builder">Paxiom model builder</param>
         /// <param name="model">Paxiom model</param>
-        /// <param name="variablesSelection">The VariablesSelection object to verify</param>
+        /// <param name="variablesSelection">The VariablesSelection object to verify and apply codelists for</param>
         /// <param name="problem">Null if everything is ok, otherwise it describes whats wrong</param>
-        /// <returns></returns>
-        public bool Verify(PXModel model, VariablesSelection? variablesSelection, out Problem? problem)
+        /// <returns>True if everything was ok, else false</returns>
+        private bool VerifyAndApplyCodelists(IPXModelBuilder builder, PXModel model, VariablesSelection? variablesSelection, out Problem? problem)
         {
             problem = null;
-
+            
             if (variablesSelection is not null && HasSelection(variablesSelection))
             {
                 //Verify that variable exists
                 foreach (var variable in variablesSelection.Selection)
                 {
-                    if (!model.Meta.Variables.Any(x => x.Code.ToUpper().Equals(variable.VariableCode.ToUpper())))
+                    Variable? pxVariable = model.Meta.Variables.FirstOrDefault(x => x.Code.ToUpper().Equals(variable.VariableCode.ToUpper()));
```

Comment:
Instead of ToUpper do a case insensitive comapre

Reply:
I was thiniking more in line with `string.Compare(x.Code, variable.VariableCode, StringComparison.InvariantCultureIgnoreCase) == 0` otherwiase we are creating allot of strings just for the comparison.